src/client.jl

```diff
@@ -34,20 +34,23 @@ convert_response(::Type{Float64}, response::T) where {T <: AbstractString} = par
 convert_response(::Type{Float64}, response::T) where {T <: Real} = float(response)::Float64
 convert_response(::Type{Bool}, response::AbstractString) = response == "OK" || response == "QUEUED" ? true : false
 convert_response(::Type{Bool}, response::Integer) = response == 1 ? true : false
-convert_response(::Type{Set{AbstractString}}, response) = Set{AbstractString}(response)
-convert_response(::Type{OrderedSet{AbstractString}}, response) = OrderedSet{AbstractString}(response)
+convert_response(::Type{Set{AbstractString}}, response::Array) = Set{AbstractString}(String(r) for r in response)
+convert_response(::Type{OrderedSet{AbstractString}}, response) = OrderedSet{AbstractString}(String(r) for r in response)
 
 function convert_response(::Type{Dict{AbstractString, AbstractString}}, response)
     iseven(length(response)) || throw(ClientException("Response could not be converted to Dict"))
     retdict = Dict{AbstractString, AbstractString}()
     for i=1:2:length(response)
-        retdict[response[i]] = response[i+1]
+        retdict[String(response[i])] = String(response[i+1])
```

Comment:
I suppose the problem here is that the function accepts `AbstractString`, but only ever returns `String`. Ideally if someone passes `Dict{SubString, SubString}` it should return that. OTOH they can always just request the binary data and do whatever themselves.